cmake/modules/Findlilv.cmake

```diff
@@ -84,5 +84,15 @@ if(lilv_FOUND)
         INTERFACE_COMPILE_OPTIONS "${PC_lilv_CFLAGS_OTHER}"
         INTERFACE_INCLUDE_DIRECTORIES "${lilv_INCLUDE_DIR}"
     )
+    get_target_property(LILV_TYPE lilv::lilv TYPE)
+    if(LILV_TYPE STREQUAL "STATIC_LIBRARY")
+      find_package(lv2 CONFIG REQUIRED)
```

Comment:
```suggestion
      find_package(lv2)
```
Are you sure `CONFIG` must be forced?

Reply:
It is taken from here: 
https://github.com/microsoft/vcpkg/blob/261c458af6e3eed5d099144aff95d2b5035f656b/ports/lilv/CMakeLists.txt#L4
and it works.

Reply:
These libraries do not ship CMake config modules. Those come from vcpkg. So I think `CONFIG REQUIRED` can only work if the static lilv library comes from vcpkg.

Reply:
The config module for lilv in vcpkg should already search for these libraries...
https://github.com/microsoft/vcpkg/blob/261c458af6e3eed5d099144aff95d2b5035f656b/ports/lilv/CMakeLists.txt#L84

Reply:
Maybe the issue is that some are linked PRIVATE?
https://github.com/microsoft/vcpkg/blob/261c458af6e3eed5d099144aff95d2b5035f656b/ports/lilv/CMakeLists.txt#L54

Reply:
Probably. And we should better not rely on transient dependencies of external libraries, even if it would be PUBLIC.

Reply:
Why shouldn't we rely on that?

Reply:
Because it to be considered an implementation detail that might change at any time.

Example: All crate dependencies in Rust are private. If I need a dependency myself I have to declare it explicitly. Cargo then will figure out if a single, shared version could be used or if multiple different versions are needed.

Reply:
We should use PUBLIC only for our own, static libs that are just an intermediate build artifact and no deployment unit.

Reply:
I disagree. Downstream applications should not be responsible for manually linking all the transient dependencies of their dependencies. IMO that indicates a bug in either the library's build system or the vcpkg port.

Reply:
Then this was a misunderstanding. Of course, not manually link unless we use transient dependencies elsewhere!